homeassistant/components/github/sensor.py

```diff
@@ -102,6 +102,7 @@ class GitHubSensorEntityDescription(BaseEntityDescription, BaseEntityDescription
         attr_fn=lambda data: {
             "sha": data["default_branch_ref"]["commit"]["sha"],
             "url": data["default_branch_ref"]["commit"]["url"],
+            "status": data["default_branch_ref"]["commit"]["status"]["state"],
```

Comment:
Given the description of this PR, this should not be an attribute.
This should be a binary sensor.

Reply:
That was my initial thought, actually, then I changed my mind when I saw the existing "latest_commit" sensor. I can change it, if that would fit better with what's already there. :+1:

Reply:
OK, so there are no binary sensors in this component yet, so adding one will be a little more involved (I'm very new here!). I'll add it and get back to you though :+1:

Reply:
Yeah, that would be a new platform for this integration, but that is where this type of information belongs, so we should do it right when introducing new data points :+1: the alternative is tacking it on here now, and then have a breaking change later, which is not good.

Reply:
Had some more time to check it out, this can actually have 5 different states, so a sensor is actually correct here.
I assumed it only failed/success

Reply:
The more I look at it, the more I actually think attribute is OK, it is actually describing the main state of this entity. :thinking: 
What do you think @timmo001 ?

Reply:
OK, I'll take it out of draft then! Still trying to find time to tick off the checklists, but if the workflow could be run, that would probably cover at least some of it.

Reply:
I think there are 2 different things.

1) Build status of the main branch
2) Build status of a commit.

As the commit sensor is representing the first, I think the build status sensor should do the same and can thus be shown as a separate sensor.

![CleanShot 2022-02-22 at 15 01 47](https://user-images.githubusercontent.com/195327/155147804-9169d856-d6d1-4e5f-9ae9-ade85c116bd3.png)

It is a very data point to show on a dashboard or automate on, for that reason, if one would automate on it, the general rule of thumb is: do not stash it in an attribute.

Reply:
Also, when exposed as its own sensor, we can provide translations for the statuses as well.

Reply:
So for home-assistant/core, this does not have a state, same with home-assistant/frontend, as this section does not include actions, of the ~10 repos I tested, the only time this have a state is for codecov

Reply:
<https://github.com/home-assistant/core/pull/66970/files#r812184527> takes care of that, now I'm with @frenck  again regarding dedicated entity :smile: